sdk/src/accounts/webSocketDriftClientAccountSubscriberV2.ts

```diff
@@ -0,0 +1,697 @@
+					_buffer: Buffer
+				) => {
+					if (
+						this.delistedMarketSetting !== DelistedMarketSetting.Subscribe &&
+						isVariant(data.status, 'delisted')
+					) {
+						return;
+					}
+					this.spotMarketAccountLatestData.set(data.marketIndex, {
+						data,
+						slot: context.slot,
+					});
+					this.eventEmitter.emit('spotMarketAccountUpdate', data);
+					this.eventEmitter.emit('update');
+				}
+			),
+			// State account subscription
+			(async () => {
+				const statePublicKey = await getDriftStateAccountPublicKey(
+					this.program.programId
+				);
+				this.stateAccountSubscriber = new WebSocketAccountSubscriberV2(
+					'state',
+					this.program,
+					statePublicKey,
+					undefined,
+					undefined,
+					this.commitment as Commitment
+				);
+				this.stateAccountSubscriber.subscribe((data: StateAccount) => {
+					this.eventEmitter.emit('stateAccountUpdate', data);
+					this.eventEmitter.emit('update');
+				});
+			})(),
+			(async () => {
+				await this.setInitialData();
+				const subscribeToOraclesStartTime = performance.now();
+				await this.subscribeToOracles();
+				const subscribeToOraclesEndTime = performance.now();
+				const duration =
+					subscribeToOraclesEndTime - subscribeToOraclesStartTime;
+				return duration;
+			})(),
+			this.stateAccountSubscriber.fetch(),
+		]);
+
+		const initialPerpMarketDataFromLatestData = new Map(
+			Array.from(this.perpMarketAccountLatestData.values()).map((data) => [
+				data.data.marketIndex,
+				data.data,
+			])
+		);
+		const initialSpotMarketDataFromLatestData = new Map(
+			Array.from(this.spotMarketAccountLatestData.values()).map((data) => [
+				data.data.marketIndex,
+				data.data,
+			])
+		);
+		this.initialPerpMarketAccountData = initialPerpMarketDataFromLatestData;
+		this.initialSpotMarketAccountData = initialSpotMarketDataFromLatestData;
+
+		this.eventEmitter.emit('update');
+
+		await this.handleDelistedMarketOracles();
+
+		await Promise.all([this.setPerpOracleMap(), this.setSpotOracleMap()]);
+
+		this.isSubscribing = false;
+		this.isSubscribed = true;
+		this.subscriptionPromiseResolver(true);
+
+		// delete initial data
+		this.removeInitialData();
+
+		const totalDuration = performance.now() - startTime;
+		console.log(
+			`[PROFILING] WebSocketDriftClientAccountSubscriberV2.subscribe() completed in ${totalDuration.toFixed(
+				2
+			)}ms`
+		);
+
+		return true;
+	}
+
+	chunks = <T>(array: readonly T[], size: number): T[][] => {
+		const result: T[][] = [];
+		for (let i = 0; i < array.length; i += size) {
+			result.push(array.slice(i, i + size));
+		}
+		return result;
+	};
+
+	public async fetch(): Promise<void> {
+		await this.setInitialData();
+	}
+
+	/**
+	 * This is a no-op method that always returns true.
+	 * Unlike the previous implementation, we don't need to manually subscribe to individual perp markets
+	 * because we automatically receive updates for all program account changes via a single websocket subscription.
```

Comment:
something that im unsure - when you mention single websocket subscription for all program account changes, this means a single WS subscription for all perp market account changes right? both spot markets and perp markets use a different websocket?

Reply:
we could technically use a single websocket for both perp and spot markets. But in this case basically yeah we use one WS connection for all perp market account changes. I think it might be fine to keep them separate. There are benefits to keep them separate like isolation if one of the connections has a problem, it doesn't affect state for the other states.